src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/AssetsComputingHelper.cs

```diff
@@ -64,6 +64,7 @@ public class AssetsComputingHelper
             ".dat" when invariantGlobalization && fileName.StartsWith("icudt") => "invariant globalization is enabled",
             ".dat" when loadFullICUData && fileName != "icudt" => "full ICU data is enabled",
             ".dat" when hybridGlobalization && fileName != "icudt_hybrid" => "hybrid globalization is enabled",
+            ".json" when hybridGlobalization && fileName != "segmentation-rules" => "hybrid globalization is enabled, expecting segmentation-rules.json",
```

Comment:
Maybe renaming function `ShouldFilterCandidate` -> `ShouldFilterOutCandidate` would make it clearer.

Reply:
If I understand the `ShouldFilterCandidate` function correctly, this condition is only triggered when HG is true to ensure that the file is named `segmentation-rules.json` (similarly with ICU cases). I'm not 100% sure if this is even necessary because the asset is specified here https://github.com/dotnet/runtime/blob/9ac537f5775853bff554051400b8988edac4dd69/src/mono/browser/build/BrowserWasmApp.targets#L113.

Reply:
The point is that this condition defines which assets will be removed, not which will be added. How are we using it:
https://github.com/dotnet/runtime/blob/1063fc278534a8b3e8f087fd3fe9c26dc03a9697/src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ComputeWasmBuildAssets.cs#L95-L100
So for HG it is not necessary, the asset would stay there, you're right. But for non-HG it would stay as well and that's why we need this line.